Fix of Issue #276

dongbohu opened the pull request:
This PR fixes issue #276. It adds paddings of `0`s into some signature names so that when they are shown on the web, the alphabetical order is consistent to the numerical order. Three files are revised:
(1) data/sample_signature_activity.txt: All signature names (on the first row) are padded.
(2) data/signature_gene_network.txt: All signature names (on the first column) are padded.
(3) data/README.md: added some description text on the padding for users who will load their own data files. I also tweaked another line in this file to make it more readable (hopefully). I hope @rzelayafavila won't mind.

dongbohu commented on data/README.md:
````diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
-  file, which is now in the `data/old/` folder and was generated by
+  based on:
+  ```
+  all-pseudomonas-gene-normalized_HWActivity_perGene_with_net300_100models_1_100_k=300_seed=123_ClusterByweighted_avgweight_network_ADAGE.txt
````
the main reason why I changed this line was because originally the super-long file name and description text were mingled together and seemed a little messy.

rzelayafavila replied:
Sure, I agree that this looks better!

dongbohu commented on data/README.md:
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
   of its size, this file has been compressed using
   [gzip](http://www.gzip.org/). However, its decompressed content is a
-  tab-delimited text file as well. 
+  tab-delimited text file as well.
```
This trailing space was corrected automatically by emacs.

dongbohu commented:
Here is a demo page that includes the new signature names:
http://165.123.67.202/#/signature/search?mlmodel=7

rzelayafavila commented on data/README.md:
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
+  which is now in the `data/old/` folder and was generated by
   [this script](https://bitbucket.org/greenelab/eadage/src/tip/node_interpretation/write_HWactivity.R).
 
+  **Note:** The signatures shown on the web (click
```
Great docs! Do you think it would be better to say "... on the web interface" or "... on the web UI" to be more specific?

dongbohu replied:
Sure. I will change "web" into "web interface".

rzelayafavila commented on data/README.md:
```diff
@@ -67,6 +76,10 @@ instance, they should be in the same format as the files below.
 
+  As we mentioned in the description of *sample_signature_activity.txt*,
+  depending on your preference of the signatures' order shown online, you may
+  need to pad some signatures' names.
```
Hmmm, I think this sentence is a little hard to follow. Maybe just changing the order and words a bit to something like "As we mentioned in the description of *sample_signature_activity.txt*, you may
need to pad some signatures' names depending on your preferred order of the signatures in the web interface."

dongbohu replied:
@rzelayafavila: Yes, I agree this sentence reads a little awkward. @mhuyck suggested another alternative which I think is great. If you also agree, I will change to his version.

rzelayafavila replied:
Sure, sounds good

rzelayafavila commented on data/sample_signature_activity.txt:
```diff
@@ -1,4 +1,4 @@
-	Sig1pos	Sig2pos	Sig3pos	Sig4pos	Sig5pos	Sig6pos	Sig7pos	Sig8pos	Sig9pos	Sig10pos	Sig11pos	Sig12pos	Sig13pos	Sig14pos	Sig15pos	Sig16pos	Sig17pos	Sig18pos	Sig19pos	Sig20pos	Sig21pos	Sig22pos	Sig23pos	Sig24pos	Sig25pos	Sig26pos	Sig27pos	Sig28pos	Sig29pos	Sig30pos	Sig31pos	Sig32pos	Sig33pos	Sig34pos	Sig35pos	Sig36pos	Sig37pos	Sig38pos	Sig39pos	Sig40pos	Sig41pos	Sig42pos	Sig43pos	Sig44pos	Sig45pos	Si ...
```
So were these padding zeroes added manually (with a Python script or something else)?

dongbohu replied:
Yes, I wrote a 4-line python script that generated the new row, then replaced the original row with it. Same idea with the other data file, but I used `cut/paste` command to replace the column.

rzelayafavila commented on data/signature_gene_network.txt:
```diff
@@ -1,102 +1,102 @@
-Sig1pos	PA0039	PA0139	PA0376	PA0409	PA0456	PA0576	PA0579	PA0762	PA0805	PA0865	PA0973	PA0996	PA1000	PA1159	PA1288	PA1544	PA1579	PA1582	PA1583	PA1588	PA1592	PA1777	PA1800	PA1801	PA1802	PA1804	PA1830	PA1847	PA2623	PA2634	PA2741	PA2742	PA2743	PA2853	PA2966	PA3031	PA3049	PA3162	PA3201	PA3202	PA3244	PA3385	PA3529	PA3533	PA3637	PA3655	PA3656	PA3745	PA3814	PA3815	PA3819	PA4067	PA4239	PA4241	PA4242	PA4243 ...
```
Also here, how was this file modified?

dongbohu replied:
I used `cut` command to copy the file w/o the first command, then created a new column with a simple Python script, then used `paste` command to clue these two parts together.

dongbohu replied:
The procedure is something like this:
```
cut -f2- signature_gene_network.txt > data_wo_col1.txt
# use python to generate a new column with filename "new_col1.txt" ...
paste new_col1.txt data_wo_col1.txt > signature_gene_network.txt
```
In `cut/paste` commands, the default delimiter is tab. (You can customize it of course.)

rzelayafavila replied:
Ok, gotcha. I was wondering if we should tell the users how we made these modifications for these files, but I guess that is up to them, as long as they know that we have modified these files.

mhuyck commented on data/README.md:
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
   [this script](https://bitbucket.org/greenelab/eadage/src/tip/node_interpretation/write_HWactivity.R).
 
+  **Note:** The signatures shown on the web (click
+  [here](http://adage.greenelab.com/#/signature/search?mlmodel=1) to see a
```
Minor quibble: could this say

```markdown
...(view [this page](http://adage.greenelab.com/#/signature/search?mlmodel=1) to
see an example)...
```

Someone sensitized me to that fact that not everyone uses a mouse anymore and it looks funny to see that written when you actually need to tap or use some assistive technology to open the link another way. Also, I think it just sounds better.

dongbohu replied:
Good point. Sure!

mhuyck commented on data/README.md:
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
 
+  **Note:** The signatures shown on the web (click
+  [here](http://adage.greenelab.com/#/signature/search?mlmodel=1) to see a
+  demo) will be in **alphabetical order**. If you would like them to be ranked
```
Saying "ranked" is technically correct, but I think "sorted" is a more common way to put this and would sound more natural.

dongbohu replied:
Hmm, originally I thought `sorted` sounds a little too "techie". I will change it to "sorted" then.

mhuyck replied:
Hah! I had the opposite impression. Either way is fine with me... it's a minor thing.

dongbohu replied:
Instead of "sorted" or "ranked", maybe we should use the more generic word "displayed".

rzelayafavila replied:
"If you would like them ordered differently..." is another option.

mhuyck commented on data/README.md:
```diff
@@ -39,18 +39,27 @@ instance, they should be in the same format as the files below.
+  **Note:** The signatures shown on the web (click
+  [here](http://adage.greenelab.com/#/signature/search?mlmodel=1) to see a
+  demo) will be in **alphabetical order**. If you would like them to be ranked
+  in another order, you may have to pad some signatures' name (such as changing
```
```
...you may have to pad the numbers in the names for some signatures...
```

mhuyck commented on data/README.md:
```diff
@@ -67,6 +76,10 @@ instance, they should be in the same format as the files below.
 
+  As we mentioned in the description of *sample_signature_activity.txt*,
+  depending on your preference of the signatures' order shown online, you may
+  need to pad some signatures' names.
```
Slightly easier to read if we cut out some extra words and order the phrases this way:

```
As mentioned in *sample_signature_activity.txt* above, you may wish to pad numbers
in signature names so they will be sorted in a useful way.
```

dongbohu replied:
How about `... in a more useful way`?

mhuyck replied:
My English teacher would have pressed me on this: more useful than *what*? If we aren't going into detail on that, I think it makes sense to caution our readers to ensure that an alphabetical sort will produce something useful.

rzelayafavila replied:
Should we include the fact that we have added this padding to these two files available in this repository (in this section and the section for *sample_signature_activity.txt* above)?

dongbohu replied:
Probably not, for two reasons:
(1) These new instructions are for users who want to load their own data, not users who will be using our data (such as Hogan Lab);
(2) The demo page already shows that we have padded our signatures.

dongbohu commented:
@rzelayafavila and @mhuyck: I think I have addressed all your comments. Please take a look again and let me know if you have any other concerns. Thanks.
